[15.0][MIG]stock_account_prepare_anglo_saxon_out_lines_hook

AaronHForgeFlow opened the pull request:
The first hook is not needed anymore, the other it is.
@JoanMForgeFlow

AaronHForgeFlow committed:
[13.0][ADD] stock_account_prepare_anglo_saxon_out_lines_hook

oca-travis committed:
[UPD] Update stock_account_prepare_anglo_saxon_out_lines_hook.pot

AaronHForgeFlow committed:
[MIG] stock_account_prepare_anglo_saxon_out_lines_hook

oca-travis committed:
[UPD] Update stock_account_prepare_anglo_saxon_out_lines_hook.pot

JoanMForgeFlow reviewed (approved):
Great! :+1:

MiquelRForgeFlow commented on stock_account_prepare_anglo_saxon_out_lines_hook/models/account_move.py:
```diff
@@ -0,0 +1,49 @@
+
+    def _prepare_interim_account_line_vals(self, line, move, debit_interim_account):
+        # Compute accounting fields.
+        sign = -1 if move.type == "out_refund" else 1
```
```suggestion
        sign = -1 if move.move_type == "out_refund" else 1
```

MiquelRForgeFlow commented on stock_account_prepare_anglo_saxon_out_lines_hook/models/account_move.py:
```diff
@@ -0,0 +1,49 @@
+
+    def _prepare_expense_account_line_vals(self, line, move, credit_expense_account):
+        # Compute accounting fields.
+        sign = -1 if move.type == "out_refund" else 1
```
```suggestion
        sign = -1 if move.move_type == "out_refund" else 1
```

MiquelRForgeFlow reviewed (commented):
please, backport this fixes to v14

AaronHForgeFlow committed:
[FIX] stock_account_prepare_anglo_saxon_out_lines_hook

MiquelRForgeFlow reviewed (approved):
I have to say that I don't see the neediness of this module, as the monkey patch now is redundant.

If you want to customize the values of `_prepare_interim_account_line_vals` and `_prepare_expense_account_line_vals`, you can already do it by calling `super` to `_stock_account_prepare_anglo_saxon_out_lines_vals`. To know if its one type or the other, you can check if they have `analytic_account_id` in the dictionary or not.

AaronHForgeFlow commented:
Yes, I can inherit from _stock_account_prepare_anglo_saxon_out_lines_vals but then I would need to do the same loop and and put all the ifs statements because the checks are the same. That would make the inheritance unefficient. And what is the drawback of adding this? Just for beauty?

AaronHForgeFlow commented:
Closing as not popular :cry:

OCA-git-bot committed:
stock_account_prepare_anglo_saxon_out_lines_hook 14.0.1.0.1

MiquelRForgeFlow committed:
[FIX] stock_account_prepare_anglo_saxon_out_lines_hook: update hook

OCA-git-bot committed:
stock_account_prepare_anglo_saxon_out_lines_hook 14.0.1.0.2

MiquelRForgeFlow committed:
[FIX] stock_account_prepare_anglo_saxon_out_lines_hook: update hook

Adapt test to https://github.com/odoo/odoo/commit/acda26fd71eac3c6bac1ad81886cd8885bc69c5a

OCA-git-bot committed:
stock_account_prepare_anglo_saxon_out_lines_hook 14.0.1.0.3

MiquelRForgeFlow committed:
[FIX] stock_account_prepare_anglo_saxon_out_lines_hook: partner on aml

Add https://github.com/odoo/odoo/commit/abfe37fcea5b20f77799d9331d4d011530880669

OCA-git-bot committed:
stock_account_prepare_anglo_saxon_out_lines_hook 14.0.1.0.4

JasminSForgeFlow committed:
[IMP] stock_account_prepare_anglo_saxon_out_lines_hook: black, isort, prettier

JasminSForgeFlow committed:
[MIG] stock_account_prepare_anglo_saxon_out_lines_hook: Migration to 15.0

AaronHForgeFlow commented:
Reopened until Odoo accepts a hook for better inheritance.

rvalyi commented:
@AaronHForgeFlow out of curiosity: do you guys use the anglo-saxon mode in Spain?

AaronHForgeFlow committed:
[MIG] stock_account_prepare_anglo_saxon_out_lines_hook to v15 (continue)

AaronHForgeFlow commented:
just created the PR to Odoo here: https://github.com/odoo/odoo/pull/98947 if that is merge this module is not needed.

Also fixed the commit history in this PR, that was incorrect.

@rvalyi  HI! :) honestly, I don't know but i don't think so. I did this for a US company and I know it is used in Denmark as well.

AaronHForgeFlow commented:
Closing, as there are a solution without the hook. The inheritance is not as easy as a prepare_method but it is possible. The module is deprecated. Sorry for the noise.

rvalyi commented:
Hello @AaronHForgeFlow that would be appreciated if you put some link to some module or even a snippet where you override the method.

MiquelRForgeFlow commented:
@rvalyi you can compare https://github.com/OCA/account-financial-tools/tree/14.0/account_move_line_sale_info with https://github.com/OCA/account-financial-tools/tree/15.0/account_move_line_sale_info as an example.

rvalyi commented:
Thanks. We are using these kind enumerate/index hacks too in some other places BTW :-)
